app/models/template.rb

```diff
@@ -31,10 +31,6 @@ def self.template_includes
     []
   end
 
-  def preview_host_collection
```

Comment:
Sure :-) yesterday I tested this PR and it works very nicely! I don't feel for the full review now but from quick scan this was the only thing that grabbed my attention.